template/registry/number.go

```diff
@@ -0,0 +1,240 @@
+There is no default value; if the style is "unit", the unit property must be provided.`,
+				ValidateValue: isPositiveInteger,
+			},
+			{
+				Name:           "unitDisplay",
+				Description:    "The unit formatting style to use in unit formatting.",
+				PossibleValues: []any{"long", "short", "narrow"},
+				Default:        "short",
+			},
+			{
+				Name: "minimumIntegerDigits",
+				Description: `The minimum number of integer digits to use.
+A value with a smaller number of integer digits than this number will be
+left-padded with zeros (to the specified length) when formatted.`,
+				ValidateValue: isPositiveInteger,
+				Default:       1,
+			},
+			{
+				Name: "minimumFractionDigits",
+				Description: `The minimum number of fraction digits to use.
+The default for plain number and percent formatting is 0;
+the default for currency formatting is the number of minor unit digits provided by
+the ISO 4217 currency code list (2 if the list doesn't provide that information).`,
+				ValidateValue: isPositiveInteger,
+			},
+			{
+				Name: "maximumFractionDigits",
+				Description: `The maximum number of fraction digits to use.
+The default for plain number formatting is the larger of minimumFractionDigits and 3;
+the default for currency formatting is the larger of minimumFractionDigits and the number of
+minor
+unit digits provided by the ISO 4217 currency code list (2 if the list doesn't provide that
+information);
+the default for percent formatting is the larger of minimumFractionDigits and 0.`,
+				ValidateValue: isPositiveInteger,
+			},
+			{
+				Name:          "minimumSignificantDigits",
+				Description:   `The minimum number of significant digits to use.`,
+				ValidateValue: isPositiveInteger,
+				Default:       1,
+			},
+			{
+				Name:          "maximumSignificantDigits",
+				Description:   `The maximum number of significant digits to use.`,
+				ValidateValue: isPositiveInteger,
+				Default:       21, //nolint:gomnd
+			},
+		},
+	},
+}
+
+// TODO: supports only style and signDisplay options.
+func numberF(input any, options map[string]any) (any, error) {
+	num, err := castAs[float64](input)
+	if err != nil {
+		return nil, fmt.Errorf("convert input to float64: %w", err)
+	}
+
+	if options == nil {
+		return num, nil
+	}
+
+	for optName := range options {
+		switch optName {
+		case "compactDisplay", "currency", "currencyDisplay", "currencySign", "notation", "numberingSystem",
+			"unit", "unitDisplay", "minimumIntegerDigits", "minimumFractionDigits",
+			"maximumFractionDigits", "minimumSignificantDigits", "maximumSignificantDigits":
+			return nil, fmt.Errorf("option '%s' is not implemented", optName)
+		}
+	}
+
+	var result string
+
+	style, ok := options["style"]
+	if !ok {
+		style = "decimal"
+	}
+
+	switch style {
+	case "decimal":
+		result = strconv.FormatFloat(num, 'f', -1, 64)
+	case "percent":
+		result = fmt.Sprintf("%.2f%%", num*100) //nolint:gomnd
```

Comment:
Does the spec imply to use 2 decimal points by default?

Reply:
for percentage yes